[BD-6] OEP-18 Compliance

ericfab179 opened the pull request:
Update setup.py install_requirements definition.
Update classifiers.
Create pip_tools, base, test, tox and travis requirements files.
Create Makefile and add upgrade command.
Include requirements files generated using upgrade command.
Remove old requirements file.
Update openedx.yml.

**Reviewers**
- [ ] @awais786 
- [ ] @morenol

edx-webhook commented:
Thanks for the pull request, @ericfab179! I've created [OSPR-4720](https://openedx.atlassian.net/browse/OSPR-4720) to keep track of it in JIRA, where we prioritize reviews.  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

Please let us know once your PR is ready for our review and all tests are green.

ericfab179 committed:
Update classifiers.
Update setup.py install_requirements definition.
Create pip_tools, base, test, tox and travis requirements files.
Create Makefile and add upgrade command.
Include requirements files generated using upgrade command.
Remove old requirements file.
Update openedx.yml.

morenol commented on requirements/base.in:
```diff
@@ -0,0 +1,11 @@
+django-sekizai
+django-mptt
+Pillow
+sorl-thumbnail
```
Could you add to the description of the PR why this dependency was added? do the same for Pillow, six and bleach.

Please also sort them alphabetically.

ericfab179 committed:
Address requirements comments.

edx-webhook commented:
@ericfab179 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/django-wiki&entry.867055334=https://github.com/edx/django-wiki/pull/61&entry.1484655318=https://github.com/ericfab179&entry.752974735=2020-06-16+04:18&entry.1917517419=2020-07-01+12:59&entry.2133058324=Yes) so we can improve your experience in the future.
